src/convert.rs

```diff
@@ -59,10 +59,20 @@ where
     type Item = A;
     type Dim = D;
     fn into_pyarray<'py>(self, py: Python<'py>) -> &'py PyArray<Self::Item, Self::Dim> {
-        let strides = self.npy_strides();
-        let dim = self.raw_dim();
-        let boxed = self.into_raw_vec().into_boxed_slice();
-        unsafe { PyArray::from_boxed_slice(py, dim, strides.as_ptr(), boxed) }
+        let (strides, dim) = (self.npy_strides(), self.raw_dim());
+        let orig_ptr = self.as_ptr();
+        let is_empty_or_size0 = self.is_empty() || std::mem::size_of::<Self>() == 0;
```

Comment:
I think this `size_of` should be for the element, `size_of::<A>` (`Self` will never be size zero).

@jturner314 and me are maintainers of rust-ndarray and don't mind reviewing changes such as these, when we have time. I'm sorry that this use case is a bit tricky, due to missing low-level APIs.

Reply:
Thank you for your kind comments. I'll fix this later.

> what you need could be a good way to improve ndarray for the use cases.

I guess @jturner314 already opened https://github.com/rust-ndarray/ndarray/pull/994?
What we need is `Box<[T]>` and the pointer to the first element of the array. The pointer is used as `void* data` in https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_NewFromDescr